test/functional/rpc_deprecated.py

```diff
@@ -30,9 +26,14 @@ def run_test(self):
         # Please ensure that for all the RPC methods tested here, there is
         # at least one other functional test that still tests the RPCs
         # functionality using the respective -deprecatedrpc flag.
+        #
+        # Please uncomment the log below when no deprecated RPC methods are being tested
+        # self.log.info("Currently no tests for deprecated RPC methods")
 
-        self.log.info("Test settxfee RPC")
-        assert_raises_rpc_error(-32, 'settxfee is deprecated and will be fully removed in v31.0.', self.nodes[0].rpc.settxfee, 0.01)
+        if self.is_wallet_compiled():
+            self.log.info("Test settxfee RPC deprecation")
+            self.nodes[0].createwallet("settxfeerpc")
```

Comment:
Is creating wallet necessary? It didn't seem to be earlier.

Reply:
Interesting, I was wondering why, ended up tracking down the conditional, makes sense now.

https://github.com/bitcoin/bitcoin/blob/c0b7159de47592c95c6db061de682b4af4058102/test/functional/test_framework/test_framework.py#L470-L471

which is due to:

https://github.com/bitcoin/bitcoin/blob/c0b7159de47592c95c6db061de682b4af4058102/test/functional/test_framework/test_framework.py#L158-L160

Reply:
The `skip_if_no_wallet` function updates the internal state of the test framework, which I wouldn't have guessed just by its name.